app/javascript/packages/analytics/is-trackable-error-event.ts

```diff
@@ -1,6 +1,7 @@
 function isTrackableErrorEvent(event: ErrorEvent): boolean {
   try {
-    return new URL(event.filename).host === window.location.host;
+    const { host, pathname } = new URL(event.filename);
+    return host === window.location.host && pathname.endsWith('.js');
```

Comment:
do you think it would make sense to add a similar check on the server? I guess we could always add later if it turns out to be too noisy

Reply:
> do you think it would make sense to add a similar check on the server? I guess we could always add later if it turns out to be too noisy

That's a good question. I think generally yes, in the sense that we shouldn't trust any "validation" to occur exclusively in the client.

But it also sparks an idea that we could do this exclusively server-side, on which I'm conflicted:

- Since this is a critical-path script, anything to reduce its size is a win
- But if we moved all validation to the server, it would put additional strain in handling X more requests, which could otherwise be filtered on the client

I'd probably lean toward keeping this as-is, and following-up with a pull request to perform the same validation server-side as a fallback.

Reply:
makes sense to me